aten/src/ATen/native/special_functions/detail/sinhc.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+namespace at {
+namespace native {
+namespace special_functions {
+namespace detail {
+template<typename T1>
+C10_HOST_DEVICE
+promote_t<T1>
+sinhc(T1 z) {
+  if (std::isnan(z)) {
+    return std::numeric_limits<T1>::quiet_NaN();
+  } else if (std::abs(z) < 4 * std::sqrt(std::numeric_limits<T1>::min())) {
+    return T1(1) + z * z / T1(6);
+  } else {
+    return std::sinh(z) / z;
+  }
+}
+}
+}
+}
+}
```

Comment:
Please add comment for namespace closing brace (and also coalesce multiple ones on the same line, similar to 
https://github.com/pytorch/pytorch/blob/c7f6deb6678f4df578584439e4ab26d185da5ef3/aten/src/ATen/native/Unfold2d.h#L30
Also, keep empty line between function/template body and namespace closing brace
```suggestion

}}}} // namespace at::native::special_functions::detail
```